Description: Enable nightly features for Cargo
 Debhelper scripts use some nightly features.
Author: Zixing Liu <zixing.liu@canonical.com>
Forwarded: not-needed
Last-Update: 2023-05-30
---
This patch header follows DEP-3: http://dep.debian.net/deps/dep3/
Index: rustc-1.70.0-src/src/tools/cargo/src/cargo/core/features.rs
===================================================================
--- rustc-1.70.0-src.orig/src/tools/cargo/src/cargo/core/features.rs
+++ rustc-1.70.0-src/src/tools/cargo/src/cargo/core/features.rs
@@ -1182,9 +1182,8 @@ pub fn channel() -> String {
             return "dev".to_string();
         }
     }
-    crate::version()
-        .release_channel
-        .unwrap_or_else(|| String::from("dev"))
+    // Debian: always return dev channel
+    String::from("dev")
 }
 
 /// Only for testing and developing. See ["Running with gitoxide as default git backend in tests"][1].
